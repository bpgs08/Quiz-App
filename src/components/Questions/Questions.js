import React from "react";
import styled from "styled-components";
import { space, media } from "../../utils/theme";
import Headings from "../../components/Headings";
import StandardButton from "../StandardButton";
import Loading from "../Loading";

let Questions = ({
  currentQuestion,
  currentQuestionContent,
  currentQuestionTitle,
  submitAnswer,
}) => {
  return (
    <QuizContainer>
      {currentQuestionTitle && currentQuestionContent ? (
        <>
          <StyledHeading type={5} bold={true} align="center">
            {currentQuestionTitle}
          </StyledHeading>
          <QuestionContainer>
            <div dangerouslySetInnerHTML={{ __html: currentQuestionContent }} />
          </QuestionContainer>

          <NumberContainer>{currentQuestion + 1} of 10</NumberContainer>
          <ButtonContainer>
            <StandardButton value="True" onClick={submitAnswer}>
              TRUE
            </StandardButton>
            <StandardButton value="False" onClick={submitAnswer}>
              FALSE
            </StandardButton>
          </ButtonContainer>
        </>
      ) : (
        <Loading>Loading...</Loading>
      )}
    </QuizContainer>
  );
};

const StyledHeading = styled(Headings)`
  margin: ${space[3]} auto ${space[3]} auto;
  max-width: 180px;
  ${media.tablet`
    max-width: 225px;
  `}
`;

const QuestionContainer = styled.div`
  width: calc(100% - 52px);
  min-height: calc(300px - 52px);
  border: 1px solid black;
  display: flex;
  align-items: center;
  padding: ${space[5]};
  text-align: center;
  margin-top: auto;
  margin-bottom: ${space[3]};
  ${media.tablet`
    min-height: calc(400px - 52px);
  `}
`;

const NumberContainer = styled.div`
  margin-top: auto;
  text-align: center;
`;

const QuizContainer = styled.div`
  margin: ${space[0]} auto;
  padding: ${space[0]} ${space[3]};
  max-width: 300px;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  justify-content: center;
  ${media.tablet`
    margin: ${space[0]} auto;
    max-width: 400px;
  `}
`;

const ButtonContainer = styled.div`
  display: flex;
  padding-bottom: ${space[8]};
  margin-top: auto;
  justify-content: space-between;
`;

export default Questions;
