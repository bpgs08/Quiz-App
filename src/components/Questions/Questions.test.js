import React from "react";
// import { BrowserRouter as Router } from "react-router-dom";
import { render, waitFor, cleanup, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Questions from "./Questions";

afterEach(cleanup);

test("should take a snapshot", () => {
  const title = "This is a Question Title";
  // true question number is questionNumber + 1
  const questionNumber = 2;
  const questionContent = `<p>content</p>`;
  const { asFragment } = render(
    <Questions
      currentQuestion={questionNumber}
      currentQuestionTitle={title}
      currentQuestionContent={questionContent}
    />
  );

  expect(
    asFragment(
      <Questions
        currentQuestion={questionNumber}
        currentQuestionTitle={title}
        currentQuestionContent={questionContent}
      />
    )
  ).toMatchSnapshot();
});

test("Renders without crashing", async () => {
  const title = "This is a Question Title";
  // true question number is questionNumber + 1
  const questionNumber = 2;
  const questionContent = `<p>content</p>`;
  const { getByText, getByLabelText } = render(
    <Questions
      currentQuestion={questionNumber}
      currentQuestionTitle={title}
      currentQuestionContent={questionContent}
    />
  );

  getByText("This is a Question Title");
  getByText(`${questionNumber + 1} of 10`);
  getByText("TRUE");
  getByText("FALSE");
});

test("Loading Screen Works", () => {
  const { getByText } = render(<Questions />);

  getByText("Loading...");
});
