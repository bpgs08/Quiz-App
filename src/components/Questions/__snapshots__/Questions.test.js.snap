// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-kEqYlL kibijz"
  >
    <h5
      class="sc-eCApGN gaEGIs sc-fujyUd fKBKxq"
      data-js-type-5="true"
    >
      This is a Question Title
    </h5>
    <div
      class="sc-pNWxx MUSPs"
    >
      <div>
        <p>
          content
        </p>
      </div>
    </div>
    <div
      class="sc-jrsJCI ezQhXB"
    >
      3 of 10
    </div>
    <div
      class="sc-iqAbSa cShxcE"
    >
      <button
        class="sc-gKAblj jzYliY"
        value="True"
      >
        TRUE
      </button>
      <button
        class="sc-gKAblj jzYliY"
        value="False"
      >
        FALSE
      </button>
    </div>
  </div>
</DocumentFragment>
`;
